### PR TITLE
added unit test for new implementation

### PR DESCRIPTION
**Pull request by annaleskiv**

### CODE REVIEW

**Comment by welps** [comment]
@bunyk @annaleskiv Did we want to merge this?

**Comment by annaleskiv** [comment]
@welps  i think yes.

**Comment by annaleskiv** [comment]
@welps  i inserted your changes here.

File: reflect.go [hunk 1]
```diff
@@ -299,6 +287,19 @@ func (r *Reflector) reflectStructFields(st *Type, definitions Definitions, t ref
 		if required {
 			st.Required = append(st.Required, name)
 		}
+
+		// return anyOf realization for structures.
```
Comment (bunyk) [inline reflect.go]: Comment is misleading

File: reflect_test.go [hunk 2]
```diff
@@ -109,3 +109,79 @@ func TestSchemaGeneration(t *testing.T) {
 		})
 	}
 }
+
+// TestUserOneOf struct
```
Comment (bunyk) [inline reflect_test.go]: This comment is useless. If linter is complaining, could we just make struct private?

**Review by bunyk — APPROVED** [review]

File: reflect.go (hunk 1 above)
Comment (annaleskiv) [reply reflect.go]: fixed

File: reflect_test.go (hunk 2 above)
Comment (annaleskiv) [reply reflect_test.go]: removed.

**Pull request merged**